features/minion/provisiond-shell/src/main/java/org/opennms/minion/provisiond/shell/MinionDetector.java

```diff
@@ -0,0 +1,142 @@
+ *      http://www.opennms.org/
+ *      http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.minion.provisiond.shell;
+
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Completion;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.opennms.netmgt.icmp.Pinger;
+import org.opennms.netmgt.icmp.PingerFactory;
+import org.opennms.netmgt.provision.SyncServiceDetector;
+
+@Command(scope = "minion", name = "detect", description = "Detects services at given ipAddress")
+@Service
+public class MinionDetector implements Action {
+
+    @Argument(index = 0, name = "detectorType", description = "Type of detector", required = true, multiValued = false, valueToShowInHelp = "icmp")
+    @Completion(DetectorTypeCompleter.class)
+    String detectorType = null;
+
+    @Argument(index = 1, name = "address", description = "The address to be detected", required = true, multiValued = false)
+    String address = null;
+
+    @Argument(index = 2, name = "attributes", description = "Attributes to be set", multiValued = true)
+    List<String> attributes;
+
+    @Reference
+    List<SyncServiceDetector> detectors;
+
+    @Reference
+    Pinger pinger;
+
+    @Override
+    public Object execute() throws Exception {
+        PingerFactory.setInstance(pinger);
+
+        InetAddress ipAddress = InetAddress.getByName(address);
+        boolean isServiceDetected = false;
+        Map<String, String> properties = parse(attributes);
+
+        for (SyncServiceDetector detector : detectors) {
+
+            if ((detector.getServiceName().equals(detectorType))) {
+                System.out.println("Trying to detect   " + detectorType
+                        + "   service at    " + address);
+                try {
+                    setAttributes(properties, detector);
+                    isServiceDetected = detector.isServiceDetected(ipAddress);
+                } catch (Exception e) {
+                    System.out.println(" Exception caused for detectorType "
+                            + detectorType);
+                }
+                System.out.println(detectorType + "   service detected at "
+                        + ipAddress + "   is   " + isServiceDetected);
+
+            }
+        }
+
+        return null;
+
+    }
+
+    private void setAttributes(Map<String, String> properties,
+            SyncServiceDetector detector) {
+
+        String ipMatch = properties.get("ipMatch");
+        String port = properties.get("port");
+        String timeout = properties.get("timeout");
+
+        detector.setIpMatch(ipMatch);
```

Comment:
We should set the attributes the same way that provisiond does when it instantiates the detectetors. See https://github.com/OpenNMS/opennms/blob/develop/opennms-provision/opennms-provisiond/src/main/java/org/opennms/netmgt/provision/service/DefaultPluginRegistry.java#L139